custom/enikshay/integrations/nikshay/repeater_generator.py

```diff
@@ -337,6 +343,102 @@ def handle_exception(self, exception, repeat_record):
                                 "followup_nikshay_registered", "followup_nikshay_error")
 
 
+@RegisterGenerator(NikshayRegisterPrivatePatientRepeater, 'case_xml', 'XML', is_default=True)
+class NikshayRegisterPrivatePatientPayloadGenerator(BaseNikshayPayloadGenerator):
+    def get_payload(self, repeat_record, episode_case):
+        person_case = get_person_case_from_episode(episode_case.domain, episode_case.get_id)
+        episode_case_properties = episode_case.dynamic_case_properties()
+        person_case_properties = person_case.dynamic_case_properties()
+
+        person_locations = get_person_locations(person_case)
+        tu_choice = person_case_properties.get('tu_choice')
+        try:
+            tu_location = SQLLocation.objects.get(location_id=tu_choice)
+        except SQLLocation.DoesNotExist:
+            raise NikshayLocationNotFound(
+                "Location with id {location_id} not found. This is the tu_choice for person with id: {person_id}"
+                .format(location_id=tu_choice, person_id=person_case.case_id)
+            )
+        try:
+            tu_code = tu_location.metadata['nikshay_code']
+        except (KeyError, AttributeError) as e:
+            raise NikshayCodeNotFound("Nikshay codes not found for location with id: {}: {}"
+                                      .format(tu_location.get_id, e))
+        episode_case_date = episode_case_properties.get('date_of_diagnosis', None)
+        if episode_case_date:
+            episode_date = datetime.datetime.strptime(episode_case_date, "%Y-%m-%d").date()
+        else:
+            episode_date = datetime.date.today()
+
+        return {
+            "Stocode": person_locations.sto,
+            "Dtocode": person_locations.dto,
+            "TBUcode": tu_code,
+            "HFIDNO": person_locations.phi,
+            "pname": person_case.name,
+            "fhname": person_case_properties.get('husband_father_name', ''),
+            "age": person_case_properties.get('age', ''),
+            "gender": person_case_properties.get('sex', '').capitalize(),
+            # API for Address with char ',' returns Invalid data format error
+            "Address": person_case_properties.get('current_address', '').replace(',', ''),
+            "pin": person_case_properties.get('current_address_postal_code', ''),
+            "lno": person_case_properties.get('phone', ''),
+            "mno": '0',
+            "tbdiagdate": str(episode_date),
+            "tbstdate": episode_case_properties.get(TREATMENT_START_DATE, str(datetime.date.today())),
+            "Type": disease_classification.get(episode_case_properties.get('pulmonary_extra_pulmonary', ''), ''),
+            "B_diagnosis": episode_case_properties.get('case_definition', ''),
```

Comment:
The case property here is `episode_case_properties.get('basis_of_diagnosis')`
The doc linked to the form where the values are. I didn't list them all out in the document since they were already listed int the form.
If you want to add extra case properties onto the case, feel free. You can call it `basis_of_diagnosis_nikshay` or something. 

Otherwise, here is the mapping:
- `clinical_chest` -> `X`
- `clinical_other` -> `E`
- `microbiological_smear` -> `S`
- `microbiological_cbnaat` -> `M`
- `microbiological_culture` -> `C`
- `microbiological_pcr` -> `M`
- `microbiological_other` -> `M`